edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test/MetricsFileStorageTest.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Moq;
+    using Xunit;
+
+    public class MetricsFileStorageTest : IDisposable
+    {
+        DateTime fakeTime;
+        ISystemTime systemTime;
+        TempDirectory tempDirectory = new TempDirectory();
+
+        public MetricsFileStorageTest()
+        {
+            var systemTime = new Mock<ISystemTime>();
+            this.fakeTime = new DateTime(100000000);
+            systemTime.Setup(x => x.UtcNow).Returns(() => this.fakeTime);
+            this.systemTime = systemTime.Object;
+        }
+
+        [Fact]
+        public void Storage()
+        {
+            string directory = this.tempDirectory.GetTempDir();
+            MetricsFileStorage storage = new MetricsFileStorage(directory, this.systemTime);
+
+            storage.WriteData(string.Join(", ", Enumerable.Range(0, 10)));
+
+            Assert.NotEmpty(Directory.GetFiles(directory));
+        }
+
+        [Fact]
+        public void GetDataSingleEntry()
+        {
+            MetricsFileStorage storage = new MetricsFileStorage(this.tempDirectory.GetTempDir(), this.systemTime);
+
+            string testData = string.Join(", ", Enumerable.Range(0, 10));
+            storage.WriteData(testData);
+
+            IDictionary<DateTime, Func<string>> actual = storage.GetData();
+            Assert.Single(actual);
+            Assert.Equal(testData, actual.Single().Value());
+        }
+
+        [Fact]
+        public void GetDataByTime()
+        {
+            MetricsFileStorage storage = new MetricsFileStorage(this.tempDirectory.GetTempDir(), this.systemTime);
+
+            storage.WriteData("data1");
+
+            IDictionary<DateTime, Func<string>> actual = storage.GetData();
+            Assert.Single(actual);
+            Assert.Equal("data1", actual.Single().Value());
+
+            DateTime break1 = this.fakeTime.AddMinutes(5);
+            this.fakeTime = this.fakeTime.AddMinutes(10);
+            storage.WriteData("data2");
+
+            actual = storage.GetData();
+            Assert.Equal(2, actual.Count);
+            actual = storage.GetData(break1);
+            Assert.Single(actual);
+            Assert.Equal("data2", actual.Single().Value());
+
+            DateTime break2 = this.fakeTime.AddMinutes(5);
+            this.fakeTime = this.fakeTime.AddMinutes(10);
+            storage.WriteData("data3");
+
+            actual = storage.GetData();
+            Assert.Equal(3, actual.Count);
+            actual = storage.GetData(break1, break2);
+            Assert.Single(actual);
+            Assert.Equal("data2", actual.Single().Value());
+        }
+
+        [Fact]
+        public void RemoveOld()
```

Comment:
Normally I will try to put the testing method as part of the test method name.
```suggestion
        public void TestRemoveOldEntries()
```